openpgp/keys.go

```diff
@@ -442,6 +452,13 @@ EachPacket:
 			current.Name = pkt.Id
 			current.UserId = pkt
 		case *packet.Signature:
+			if pkt.SigType == packet.SigTypeKeyRevocation {
+				// These revocations won't revoke UIDs (see
+				// SigTypeIdentityRevocation). Handle these first,
+				// because key might have revocation coming from
+				// another key (designated revoke).
+				revocations = append(revocations, pkt)
```

Comment:
I.e., we're also appending this onto `current.Sigantures`, whereas previous we weren't? Do I read that properly and is that OK?